### DISCUSSION TITLE
targets commands without functions

### DISCUSSION THREAD

**Discussion by llrs** [ropensci/targets#779] [General]
Recently I observed that the documentation uses/recommends always functions like [this example](https://books.ropensci.org/targets/files.html#external-input-files):

```r
# _targets.R
library(targets)
path_to_data <- function() {
  "data/raw_data.csv"
}
list(
  tar_target(
    raw_data_file,
    path_to_data(),
    format = "file"
  ),
  tar_target(
    raw_data,
    read_csv(raw_data_file, col_types = cols())
  )
)
```

Perhaps it is me misusing a feature of R but, it is possible to define and use easier code:

```r

# _targets.R
library(targets)

list(
  tar_target(
    raw_data_file,
    "data/raw_data.csv",
    format = "file"
  ),
  tar_target(
    raw_data,
    read_csv(raw_data_file, col_types = cols())
  )
)
```
Are there any negative consequences of this ? It probably goes against the good programming practices but makes it easier later on to create sub-functions. (I've also used for more complicated code like using {} within the command to have something like mini-script/function inside it).

Sorry I haven't found anything on the book or help page (probably for the reason I mentioned) but I'm curious.

**Comment by wlandau** [comment]
I do intend `targets` to be this flexible. This is where writing code gets to be more of an art than a science, and specific cases are up to the judgement of the user. My usual recommendation is to use concise function calls for the vast majority of targets, but it is possible to take modularity too far. If you have 20 different plots at the end of your pipeline, for example, a function for every single one might seem excessive.

**Reply by llrs** [reply]
Great to know there aren't any downsides of this approach. This might help to beginners of targets to move from a script to a targets pipeline (instead of starting from 0. This might be an easy way to do it (as intermediate step).
